### PR TITLE
repo sync

### PR DESCRIPTION
**Pull request by Octomerger**
This is an automated pull request to sync changes between the public and private repos.

:robot: This pull request should be merged (not squashed) to preserve continuity across repos, so please let a bot do the merging!

### CODE REVIEW

**Review by janik302 — DISMISSED** [review]

**Comment by prince582** [comment]
Ok

**Review by mpx412 — APPROVED** [review]

**Review by mpx412 — APPROVED** [review]

**Review by mpx412 — APPROVED** [review]

**Comment by MarkPhillips00763** [comment]
MarkPhillips00763@yahoo.com

**Review by mpx412 — APPROVED** [review]

**Review by mpx412 — DISMISSED** [review]

File: content/codespaces/about-codespaces.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
+---
+title: About Codespaces
+intro: '{% data variables.product.prodname_codespaces %} is a configurable online development environment, hosted by {% data variables.product.prodname_dotcom %} and powered by {% data variables.product.prodname_vscode %}, that allows you to develop entirely in the cloud.'
```
Comment (SinEstresOrrantia) [inline content/codespaces/about-codespaces.md:3]:
```
> ```suggestion
intro: '{% data variables.product.prodname_codespaces %} is a configurable online development environment, hosted by {% data variables.product.prodname_dotcom %} and powered by {% data variables.product.prodname_vscode %}, that allows you to develop entirely in the cloud.'
```
If Music Label=Name
    Print(Strezless Musick Productionz) 
If Artist=Name
     Print(Sin Estres)
```

Comment (SinEstresOrrantia) [reply content/codespaces/about-codespaces.md:3]: +Omar Dejesu Orrantia

File: content/codespaces/about-codespaces.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+product: '{% data reusables.gated-features.codespaces %}'
```
Comment (SinEstresOrrantia) [inline content/codespaces/about-codespaces.md:4]: > +digital music
   `No Le Salen`
> +Single track
  `  No le salen`

File: content/codespaces/about-codespaces.md [hunk 3]
```diff
@@ -0,0 +1,72 @@
+---
+title: About Codespaces
```
Comment (SinEstresOrrantia) [inline content/codespaces/about-codespaces.md:2]: Strezless Musick by Sin Estres Orrantia

File: content/codespaces/about-codespaces.md (hunk 1 above)
Comment (SinEstresOrrantia) [inline content/codespaces/about-codespaces.md:3]: Strezless Musick presents Sin Estres

File: content/codespaces/about-codespaces.md [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+intro: '{% data variables.product.prodname_codespaces %} is a configurable online development environment, hosted by {% data variables.product.prodname_dotcom %} and powered by {% data variables.product.prodname_vscode %}, that allows you to develop entirely in the cloud.'
+product: '{% data reusables.gated-features.codespaces %}'
+redirect_from:
```
Comment (SinEstresOrrantia) [inline content/codespaces/about-codespaces.md:5]: Search music engines or Distributors

File: content/codespaces/setting-up-your-codespace/configuring-codespaces-for-your-project.md [hunk 5]
```diff
@@ -1,8 +1,8 @@
 ---
 title: Configuring Codespaces for your project
-intro: You can set up a default configuration for every new codespace for your repository to ensure that contributors have all the tools and settings they need in their online development environment.
+intro: You can use a `devcontainer.json` file to define a {% data variables.product.prodname_codespaces %} environment for your repository. 
 product: '{% data reusables.gated-features.codespaces %}'
```
Comment (SinEstresOrrantia) [inline content/codespaces/setting-up-your-codespace/configuring-codespaces-for-your-project.md:4]: for ( index = 0 to input.length-1 ) {

				bool sorted = true

				for ( j = 0 to j < input.length-1-index ) {

						if ( input[ j ] > input[ j+1 ] ) {

								swap input[ j+1 ] & input[ j ]

								sorted = false

						}

				}

				// Stop, once the array is sorted
				if ( sorted )
						return;
		}

**Review by SinEstresOrrantia — APPROVED** [review]
Pass

File: content/codespaces/codespaces-reference/about-billing-for-codespaces.md [hunk 6]
```diff
@@ -6,6 +6,7 @@ versions:
   free-pro-team: '*'
```
Comment (SinEstresOrrantia) [inline content/codespaces/codespaces-reference/about-billing-for-codespaces.md:6]: https://youtube.com/c/SinEstresOrrantia

https://distrokid.com/hyperfollow/sinestres/not-another-broken-heart

https://distrokid.com/hyperfollow/sinestres/hablan-mal-de-mi

Check out Omar Orrantia’s profile on LinkedIn https://www.linkedin.com/in/omar-orrantia-13032999

https://distrokid.com/hyperfollow/sinestres/do-you-like-me	


```suggestion
  - /github/developing-online-with-github-codespaces/about-github-codespaces
```
	for ( i = 0 to i = input.length-1 ) {

				int temp = input[ i ]
				boolean hasMoved = false
				boolean notPlaced = true

				for ( j = i-1 to j = 0 ) {

						if ( input[ j ] < temp ) {
								//  Shift numbers till you find it’s right location
								input[ j+1 ] = input[ j ]
								hasMoved = true
						} else {
								input[ j+1 ] = temp
								notPlaced = false
								break;
						}

				}

				// Edge case
				if( hasMoved && notPlaced ) 
						input[0] = temp 

		}

Comment (SinEstresOrrantia) [inline content/codespaces/codespaces-reference/about-billing-for-codespaces.md:6]: [Sin Estres Wix](https://therealsinestres.wixsite.com/sinestres)

Comment (SinEstresOrrantia) [inline content/codespaces/codespaces-reference/about-billing-for-codespaces.md:6]: [_Omar Dejesu Orrantia is founder of music publishing label=Strezless Musick Productionz_](https://youtube.com/c/SinEstresOrrantia)

File: content/codespaces/about-codespaces.md [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+  - /github/developing-online-with-github-codespaces/about-github-codespaces
```
Comment (SinEstresOrrantia) [inline content/codespaces/about-codespaces.md:6]: quicksort ( int[] input, int left, int right ) {

				if ( right - left <= 0 )
						return;

				int pivot = input[ ( left + right ) / 2 ];
				int i = left,  j = right;

				while ( i <= j ) {
						while ( input[ i ] < pivot )
								i++;

						while ( input[ j ] > pivot )
								j--;

						if ( i <= j ) {
								swap input[ i ] & input[ j ]
								i++;  j--;
						}
				}

				// Recursive calls
				quicksort ( input, left, i-1 );
				quicksort ( input, i, right );
		}

**Review by SinEstresOrrantia — APPROVED** [review]

File: content/codespaces/setting-up-your-codespace/configuring-codespaces-for-your-project.md [hunk 8]
```diff
@@ -1,8 +1,8 @@
 ---
 title: Configuring Codespaces for your project
```
Comment (SinEstresOrrantia) [inline content/codespaces/setting-up-your-codespace/configuring-codespaces-for-your-project.md:2]: Strezless Musick Productionz

Comment (SinEstresOrrantia) [inline content/codespaces/setting-up-your-codespace/configuring-codespaces-for-your-project.md:2]: Omar Dejesu Orrantia

Comment (SinEstresOrrantia) [inline content/codespaces/setting-up-your-codespace/configuring-codespaces-for-your-project.md:2]: Sin Estres_tag_idx_Strezless_Musick_Productionz

File: content/codespaces/setting-up-your-codespace/configuring-codespaces-for-your-project.md [hunk 9]
```diff
@@ -1,8 +1,8 @@
...
-intro: You can set up a default configuration for every new codespace for your repository to ensure that contributors have all the tools and settings they need in their online development environment.
```
Comment (SinEstresOrrantia) [inline content/codespaces/setting-up-your-codespace/configuring-codespaces-for-your-project.md:3]: - [https://therealsinestres.wixsite.com/sinestres]

File: content/codespaces/setting-up-your-codespace/configuring-codespaces-for-your-project.md (hunk 5 above)
Comment (SinEstresOrrantia) [inline content/codespaces/setting-up-your-codespace/configuring-codespaces-for-your-project.md:4]: [Hanlan mal de mi](Hhttps://distrokid.com/hyperfollow/sinestres/hablan-mal-de-mi)

File: content/codespaces/index.md [hunk 10]
```diff
@@ -16,7 +16,9 @@ topics:
 
 ### Table of Contents
```
Comment (SinEstresOrrantia) [inline content/codespaces/index.md:18]: Coming soon


```suggestion
 https://distrokid.com/hyperfollow/sinestres/not-another-broken-heart
```
https://distrokid.com/hyperfollow/sinestres/hablan-mal-de-mi

File: content/codespaces/setting-up-your-codespace/configuring-codespaces-for-your-project.md [hunk 11]
```diff
@@ -14,27 +14,66 @@ topics:
...
+
+### Using the default configuration
+
+If you don't define a configuration in your repository, {% data variables.product.prodname_dotcom %} creates a codespace with a base Linux image. The base Linux image includes languages and runtimes like Python, Node.js, JavaScript, TypeScript, C++, Java, .NET, PHP, PowerShell, Go, Ruby, and Rust. It also includes other developer tools and utilities like git, GitHub CLI, yarn, openssh, and vim. To see all the languages, runtimes, and tools that are included use the `devcontainer-info content-url` command inside your codespace terminal and follow the url.
+
+For more information about everything that is included in the base Linux image, see the latest file in the [`microsoft/vscode-dev-containers`](https://github.com/microsoft/vscode-dev-containers/tree/master/containers/codespaces-linux/history) repository. 
+
+The default configuration is a good option if you're working on a small project that uses the languages and tools that {% data variables.product.prodname_codespaces %} provides.
```
Comment (SinEstresOrrantia) [inline content/codespaces/setting-up-your-codespace/configuring-codespaces-for-your-project.md:67]: Strezless Musick Productionz 


```suggestion
The default configuration is a good option if you're working on a small project that uses the languages and tools that {% data variables.product.prodname_codespaces %} provides.
Strezless Musick Productionz
-**Artist=**Sin**-Estres**
-https://distrokid.com/hyperfollow/sinestres/hablan-mal-de-mi
-![VideoCapture_20210405-120517.jpg](https://user-images.githubusercontent.com/78077841/117613087-ee9e6480-b12b-11eb-91f6-569206bb53e8.jpg)

```

File: content/codespaces/setting-up-your-codespace/configuring-codespaces-for-your-project.md [hunk 12]
```diff
@@ -14,27 +14,66 @@ topics:
...
+```
+
+For more information on using a Dockerfile in a dev container, see [](https://code.visualstudio.com/docs/remote/create-dev-container#_dockerfile) in the {% data variables.product.prodname_vscode %} documentation. 
+
+### Using the default configuration
+
+If you don't define a configuration in your repository, {% data variables.product.prodname_dotcom %} creates a codespace with a base Linux image. The base Linux image includes languages and runtimes like Python, Node.js, JavaScript, TypeScript, C++, Java, .NET, PHP, PowerShell, Go, Ruby, and Rust. It also includes other developer tools and utilities like git, GitHub CLI, yarn, openssh, and vim. To see all the languages, runtimes, and tools that are included use the `devcontainer-info content-url` command inside your codespace terminal and follow the url.
+
+For more information about everything that is included in the base Linux image, see the latest file in the [`microsoft/vscode-dev-containers`](https://github.com/microsoft/vscode-dev-containers/tree/master/containers/codespaces-linux/history) repository. 
+
```
Comment (SinEstresOrrantia) [inline content/codespaces/setting-up-your-codespace/configuring-codespaces-for-your-project.md:66]: STREZLESS MUSICK PRODUCTIONZ

File: content/codespaces/setting-up-your-codespace/configuring-codespaces-for-your-project.md [hunk 13]
```diff
@@ -14,27 +14,66 @@ topics:
...
+The default configuration is a good option if you're working on a small project that uses the languages and tools that {% data variables.product.prodname_codespaces %} provides.
+
+
+### Using a predefined container configuration
+
+Predefined container definitions include a common configuration for a particular project type, and can help you quickly get started with a configuration that already has the appropriate container options, {% data variables.product.prodname_vscode %} settings, and {% data variables.product.prodname_vscode %} extensions that should be installed.
+
+Using a predefined configuration is a great idea if you need some additional extensibility. You can also start with a predefined configuration and amend it as needed for your project's setup. 
+
+1. To access the command palette, in the upper-left corner, select the Application Menu and click **Command Palette…** from the **View** menu, then start typing "Codespaces: Add Development Container Configuration Files...". Click **Codespaces: Add Development Container Configuration Files...**
```
Comment (SinEstresOrrantia) [inline content/codespaces/setting-up-your-codespace/configuring-codespaces-for-your-project.md:76]: Strezless Musick

**Review by SinEstresOrrantia — APPROVED** [review]
I think life is amazing

**Review by adamschris54321 — DISMISSED** [review]

**Review by atheerhayyan — DISMISSED** [review]

**Review by leocedeno — COMMENTED** [review]
Thank you

**Pull request merged**